### PR TITLE
Add support for effects, transition/brightness parameters to template light, min_mireds and max_mireds templates

### PR DESCRIPTION
**Pull request by jacekpaszkowski**
## Proposed change

Documetation update accordingly to https://github.com/home-assistant/core/pull/43850

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/43850
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by jacekpaszkowski** [comment]
could you please tell me what is the best and safe method to rebase my forked branch?

**Review by frenck — CHANGES_REQUESTED** [review]
This PR targets an incorrect branch. As this depends on a next release of Home Assistant, it should be targeting the `next` branch.

Please change the target branch to `next` and git rebase your PR if needed.

File: source/_integrations/light.template.markdown [hunk 1]
```diff
@@ -145,6 +179,8 @@ For example, you would replace
 with this equivalent that returns `true`/`false` and never gives an unknown
 result:
 {% raw %}`{{ is_state('switch.source', 'on') }}`{% endraw %}
+Passing {% raw %}`transition`{% endraw %} and {% raw %}`brightness`{% endraw %} parameters along with {% raw %}`*_color`{% endraw %}, {% raw %}`color_temp`{% endraw %}, {% raw %}`white_value`{% endraw %} or {% raw %}`effect`{% endraw %} is supported.
```
Comment (frenck) [inline source/_integrations/light.template.markdown]: These do not have to be escaped, thus all these raw/endraw's can be removed.

**Review by frenck — CHANGES_REQUESTED** [review]

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Comment by emontnemery** [comment]
Please make it clear that:
- Transition doesn't have its own script, it will instead be passed as a parameter to the brightness, color_temp, effect, hs_color or white_value scripts
- Brightness in a turn_on call will be passed as a parameter to either of color_temp, effect, hs_color or white_value scripts if the corresponding parameter is also in the turn_on call. In this case the brightness script will not be called.

File: source/_integrations/light.template.markdown [hunk 2]
```diff
@@ -99,6 +115,26 @@ light:
         required: false
         type: template
         default: optimistic
+      supports_transition_template:
+        description: Defines a template to get if light supports transition.
```
Comment (emontnemery) [inline source/_integrations/light.template.markdown]: Explain also that:
- What the template should return (True/False?)
- That a transition in a turn on or turn off call will be passed as a named parameter "transition" to either of the scripts.

File: source/_integrations/light.template.markdown [hunk 3]
```diff
@@ -145,6 +185,8 @@ For example, you would replace
 with this equivalent that returns `true`/`false` and never gives an unknown
 result:
 {% raw %}`{{ is_state('switch.source', 'on') }}`{% endraw %}
+Passing `transition` and `brightness` parameters along with `*_color`, `color_temp`, `white_value` or `effect` is supported.
```
Comment (emontnemery) [inline source/_integrations/light.template.markdown]: Explain that "transition" will be passed as named parameter `transition` to the scripts.
Explain that "brightness" will be passed as named parameter `brightness` to the scripts.

Please also update the documentation for `set_level` to explain that the script will only be called if the `turn_on` call only has brightness, and optionally, transition.

File: source/_integrations/light.template.markdown (hunk 2 above)
Comment (jacekpaszkowski) [reply source/_integrations/light.template.markdown]: done

File: source/_integrations/light.template.markdown (hunk 3 above)
Comment (jacekpaszkowski) [reply source/_integrations/light.template.markdown]: done

**Review by emontnemery — APPROVED** [review]

**Review by frenck — APPROVED** [review]
Thanks, @jacekpaszkowski 👍

**Pull request merged**